crates/oxc_semantic/src/lib.rs

```diff
@@ -285,8 +286,8 @@ mod tests {
             (SourceType::default(), "let a = 1, b; b = a += 5", ReferenceFlag::read_write()),
             (SourceType::default(), "let a = 1; a += 5", ReferenceFlag::read_write()),
             // note: we consider a to be written, and the read of `1` propagates upwards
-            (SourceType::default(), "let a, b; b = a = 1", ReferenceFlag::write()),
-            (SourceType::default(), "let a, b; b = (a = 1)", ReferenceFlag::write()),
+            (SourceType::default(), "let a, b; b = a = 1", ReferenceFlag::read_write()),
+            (SourceType::default(), "let a, b; b = (a = 1)", ReferenceFlag::read_write()),
```

Comment:
After careful analysis, I think these changes are expected.

You can see this case. `(a = 1)` is write, `b = a` is read. So `a` has read and write reference flag 
```ts
let a;
b;
b = (a = 1)
```